### PR TITLE
Feature/43693 port school import to typescript

### PR DESCRIPTION
**Pull request by GuyHarwood**
Migrate `school-import` from `functions-app` to `tslib` to run under `func-throttled` app

- [x] port javascript to typescript
- [x] extract mapping and filters to separate classes
- [x] improve logging & output capture

### CODE REVIEW

File: tslib/src/functions-throttled/school-import/predicates.ts
Comment (jon-shipley) [inline tslib/src/functions-throttled/school-import/predicates.ts:55]: This magic number seems odd.  "704".

**Review by jon-shipley — COMMENTED** [review]
LGSF

File: tslib/src/functions-throttled/school-import/predicates.ts
Comment (GuyHarwood) [reply tslib/src/functions-throttled/school-import/predicates.ts:55]: you wrote the original code 😄 
in the census import PR i turned this into a const, which i intend to do for this also, once the census import is merged in.

its for schools in gibraltar.

Comment (jon-shipley) [reply tslib/src/functions-throttled/school-import/predicates.ts:55]: I'm thinking that this would be modelled better as an exclusion list, but I'll leave it up to you.

File: tslib/src/functions-throttled/school-import/school-import.service.spec.ts
Comment (GuyHarwood) [inline tslib/src/functions-throttled/school-import/school-import.service.spec.ts:29]: to be completed

File: tslib/src/functions-throttled/school-import/data-access/school.data.service.ts
Comment (jon-shipley) [inline tslib/src/functions-throttled/school-import/data-access/school.data.service.ts:37]: Why is this col (and leaCode) nullable?

File: tslib/src/functions-throttled/school-import/data-access/school.data.service.ts
Comment (jon-shipley) [inline tslib/src/functions-throttled/school-import/data-access/school.data.service.ts:50]: If the input does not guarantee the `estabCode` and `leaCode` then perhaps a comment would help guide the reader about the input format irregularities?  If that's what is actually happening here.

File: tslib/src/functions-throttled/school-import/school-import.service.ts
Comment (jon-shipley) [inline tslib/src/functions-throttled/school-import/school-import.service.ts:108]: Can you explain this error message?

**Review by jon-shipley — APPROVED** [review]
LGTM, but some non-blocking comments raised.

File: tslib/src/functions-throttled/school-import/data-access/school.data.service.ts
Comment (GuyHarwood) [reply tslib/src/functions-throttled/school-import/data-access/school.data.service.ts:37]: untouched from original implementation, good spot though

File: tslib/src/functions-throttled/school-import/school-import.service.ts
Comment (GuyHarwood) [reply tslib/src/functions-throttled/school-import/school-import.service.ts:108]: No 🤣 

there was a `test.todo` in the spec, but i overlooked this.  will sort today

File: tslib/src/functions-throttled/school-import/data-access/school.data.service.ts
Comment (GuyHarwood) [reply tslib/src/functions-throttled/school-import/data-access/school.data.service.ts:37]: leaCode & estabCode are nullable in the database, which seems odd given that dfeNumber is not.  No schools have, nor should have a null value for either of these.  Will create a ticket for that, this will have to remain as-is for now and will be remediated in the upcoming ticket.

File: tslib/src/functions-throttled/school-import/data-access/school.data.service.ts
Comment (GuyHarwood) [reply tslib/src/functions-throttled/school-import/data-access/school.data.service.ts:52]: this is old code too, comment added though

**Pull request merged**
